examples/animation/animated_fox.rs

```diff
@@ -17,19 +19,51 @@ fn main() {
             brightness: 2000.,
         })
         .add_plugins(DefaultPlugins)
+        .init_resource::<ParticleAssets>()
+        .init_resource::<FoxFeetTargets>()
         .add_systems(Startup, setup)
         .add_systems(Update, setup_scene_once_loaded.before(animate_targets))
-        .add_systems(Update, keyboard_animation_control)
+        .add_systems(Update, (keyboard_animation_control, simulate_particles))
+        .observe(OnStep::observer)
         .run();
 }
 
 #[derive(Resource)]
 struct Animations {
     animations: Vec<AnimationNodeIndex>,
-    #[allow(dead_code)]
     graph: Handle<AnimationGraph>,
 }
 
+#[derive(Event, AnimationEvent, Reflect, Clone)]
+#[reflect(AnimationEvent)]
+struct OnStep;
+
+impl OnStep {
+    fn observer(
```

Comment:
```suggestion
    // The observer is not required to be defined in an impl block
    fn observer(
```

Small suggestion since there is no strong connection between the observer function and the type (ie: it doesn't take self, etc so you can't call it on an OnStep value, its for code organization). Newer programmers may find this pattern confusing and think it *has* to be in an impl block because of a lack of understanding the derive/reflects in play.

Alternatively it could be defined as a standalone function, for example: `fn observe_on_step`.

Reply:
good idea, removed in favor of `observe_on_step`.